### DISCUSSION TITLE
Disable AVX / AVX2 on Openssl

### DISCUSSION THREAD

**Discussion by quantropi-RYANTOTH** [open-quantum-safe/openssl#371] [Q&A] [answered]
Hello,

I'm trying to disable AVX / AVX2 instructions. I've researched online and throughout the repo, and found the following command to try but it didn't seem to work after executing it and recompiling.

`env OPENSSL_ia32cap=:0 apps/openssl`

I'm looking for any suggestions that might help me achieve this functionality. Thank you!

**Comment by baentsch** [comment, marked as answer]
This looks like a general (upstream) OpenSSL question. Further looking at the command above, the ":" seems odd: Are you sure that syntax is right? Lastly, if you compile OpenSSL on a platform without AVX(2) support, those instructions should not be compiled in, right?

**Reply by quantropi-RYANTOTH** [reply]
Thanks for the quick response -- I went ahead and asked the question in OpenSSL. 

You are right in that if the platform doesn't support AVX(2), it won't be executed.

The solution I used was just commenting out the AVX(2) calling functions in liboqs (since my machine had AVX(2) enabled).

**Reply by baentsch** [reply]
Just out of curiosity: What is the reason you'd like to disable AVX(2) if your machine supports it? If you'd like to create "distributable" `liboqs` object code (e.g., in docker images) then you probably simply want to set [OQS_DIST_BUILD](https://github.com/open-quantum-safe/liboqs/wiki/Customizing-liboqs#OQS_DIST_BUILD)=ON and [OQS_OPT_TARGET](https://github.com/open-quantum-safe/liboqs/wiki/Customizing-liboqs#oqs_opt_target)=generic when running `cmake`.
